### ISSUE TITLE
Training w/ default config diverges and ultimately crashes: NaN or Inf found in input tensor

### ISSUE THREAD

**Issue by alexsax** [OpenGVLab/PonderV2#10]
Hi! Thank you for this work and for the well-documented code release!

I am trying to reproduce the pretraining results, and I am seeing training diverge quite quickly, within 5 batches when using the default config. Is this expected, and how have you dealt with it in the past?

In particular I will see some losses be nan intermittently:
```
[2024-02-14 04:57:41,686 INFO misc.py line 120 1075918] Train: [2/100][82/150] Data 0.214 (0.115) Batch 28.240 (31.931) Remain 130:59:20 loss: nan depth_loss: nan rgb_loss: nan psnr: nan semantic_loss: 0.8146 free_space_loss: nan sdf_loss: nan eikonal_loss: nan Lr: 0.00009
NaN or Inf found in input tensor.
...
NaN or Inf found in input tensor.
[2024-02-14 04:58:19,912 INFO misc.py line 120 1075918] Train: [2/100][83/150] Data 0.025 (0.114) Batch 38.136 (32.009) Remain 131:17:54 loss: 3.1007 depth_loss: 0.1920 rgb_loss: 0.9712 psnr: 14.9213 semantic_loss: 0.8159 free_space_loss: 0.1176 sdf_loss: 0.7030 eikonal_loss: 0.3010 Lr: 0.00009
```

For reference, I am running the following command to pretrain on just scannet using 8 v100s:
```
bash scripts/train.sh -g 8 -d scannet -c pretrain-ponder-spunet-v1m1-0-base -n ponderv2-pretrain-sn-base
```

I was able to reproduce other results outside of pretraining: that is, I have been able to successfully evaluate the pretrained weights: `ponderv2-ppt-ft-semseg-scannet.pth`, and to finetune `ponderv2-ppt-pretrain-scannet-s3dis-structured3d.pth` and achieve 0.77 mIoU on ScanNet semseg.

**Comment by alexsax** [comment]
And eventually training will crash with a shape error:
```
File "/private/home/ssax/accel-cortex/pretrain/exp/scannet/ponderv2-pretrain-sn-base/code/ponder/models/ponder/ponder_indoor_base.py", line 688, in forward
    ray_dict, data_dict = self.prepare_ray(data_dict)
  File "/private/home/ssax/miniconda3/envs/ponderv2/lib/python3.8/site-packages/torch/autograd/grad_mode.py", line 27, in decorate_context
    return func(*args, **kwargs)
  File "/private/home/ssax/accel-cortex/pretrain/exp/scannet/ponderv2-pretrain-sn-base/code/ponder/models/ponder/ponder_indoor_base.py", line 624, in prepare_ray
    ray_dict = self.ray_sample(data_dict)
  File "/private/home/ssax/miniconda3/envs/ponderv2/lib/python3.8/site-packages/torch/autograd/grad_mode.py", line 27, in decorate_context
    return func(*args, **kwargs)
  File "/private/home/ssax/accel-cortex/pretrain/exp/scannet/ponderv2-pretrain-sn-base/code/ponder/models/ponder/ponder_indoor_base.py", line 595, in ray_sample
    batch_data[k] = torch.stack([v[k] for v in view_data], dim=0)
RuntimeError: stack expects each tensor to be equal size, but got [256, 3] at entry 0 and [156, 3] at entry 2
```

Any help debugging these issues is much appreciated :)

**Comment by alexsax** [comment]
Fixed by #11
